uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/App.java

```diff
@@ -108,6 +127,21 @@ public void launch() {
         CommonUtils.launchApp(packageName);
     }
 
+    @Override
+    public void update () {
+        if (updateAppInstaller instanceof LocalApkInstaller && !TextUtils.isEmpty(localUpdateApkFileName)) {
+            Logger.i(TAG, "Installing the " + this.appName + " from local apk..");
+            updateAppInstaller.installApp(localUpdateApkFileName);
+        } else {
+            Logger.i(TAG, "Installing the " + this.appName + " from Play Store..");
+            updateAppInstaller.installApp(packageName);
+        }
+
+        // the app is just installed, first run should be handled
+        // this value can (should) be changed to false by child classes as appropriate
+        shouldHandleFirstRun = true;
```

Comment:
Is this actually required for update?

Reply:
I tested the update scenario and this is not required as first run flag is stored in storage in authenticator and this is not reset on update. Hence removing this piece of code